### ISSUE TITLE
Workflow or Saga

### ISSUE THREAD

**Issue by refactorthis** [justeat/JustSaying#318]
Hi,

How do you go about doing workflows?

**Comment by AnthonySteele** [comment]
Could you give more detail on what you mean by workflows? Which AWS features are involved, or is more of a general concept?

**Comment by refactorthis** [comment]
I guess I should have elaborated a bit. For a workflow I could perhaps use SWF. I'm more looking for a eventually consistent distributed transactions like this:

http://masstransit-project.com/MassTransit/advanced/sagas/ 

https://docs.particular.net/nservicebus/sagas/

**Comment by AnthonySteele** [comment]
I expect that making a fully working consistent distributed transaction co-coordinator is well out of scope of JustSaying. If anyone does attempt it, I would recommend that it be a layer over JustSaying. And that people understand that this is not an easy or simple thing to do well.

**Comment by eric-davis** [comment]
I would agree with Anthony's assessment. I have recently started down this
path. Creating what I am calling domain processes and process managers
(saga term is over-used and a bit overloaded). These are specialized domain
models that include JustSaying message handlers, publish events, submit
commands, act as a sort of state machine, and persist events in a basic
event store for re-hydrating these models to get current state of the
process.

Seems to be working for what I intended so far...but still at the beginning
of this effort.
